packages/contracts/src/interfaces/Extension.sol

```diff
@@ -24,8 +24,10 @@ abstract contract Extension {
 
     /// Register unclaimed state for a recipient emailCommitment
     /// @param unclaimedState Unclaimed state that is registered
-    function registerUnclaimedState(UnclaimedState memory unclaimedState) public virtual returns (bool) {
+    /// @param isInternal A flag whether the unclaimed state is registered from `registerUnclaimedStateAsExtension`.
```

Comment:
For example, NFT extension sends NFT to an email address by writing a token id in an unclaimed state. It will use the registerUnclaimedState to lock the sender's NFT in the extension contract to prevent the sender from sending the NFT before the unclaimed state is claimed/reverted.

Reply:
Yea, but why `isInternal` is required? The extension just need to lock the tokenId on `registerUnclaimedState` - the bhaviour would be same if it was called internally or externally.
Even if some extra logic is needed for internal, extension can do it in the execute after calling core.registerUS

Reply:
> the bhaviour would be same if it was called internally or externally.
When the extension is called by the email wallet user, this is right.
However, when an external contract registers an unclaimed state to send NFT to a specific email address, I think there is no way to ensure that the registered NFT is locked.

Reply:
In this case external contract set allowance for the tokenId and call registerUS on core which trigger registerUS on extension - this method move the token to extension contract.
The state bytes in US need to be encoded in a way that extension understand anyway.

Reply:
> In this case external contract set allowance for the tokenId and call registerUS on core which trigger registerUS on extension - this method move the token to extension contract.

Yes, this is why the extension needs to implement registerUS.

> The state bytes in US need to be encoded in a way that extension understand anyway.

Yes, so each extension will define a type of the unclaimed state, and the caller passes `abi.encode` of that type daya.

Reply:
Sorry for the confusion. I was telling `isInternal` arg is not required in the above case.
Do we need to pass `isInternal` function arg to `extension.execute()`? Its ok either way.

Reply:
This is a problem of developer experience (clarity of the API interface).
Could you remove `isInternal` arg and rename the function name to `registerUnclaimedStateExternal` in the extension interface?

Reply:
^ But core would be calling the same function when an external contract registerUS also.
I think we can keep the `isInternal` flag. Its not a issue.